pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/PdfDao.java

```diff
@@ -510,27 +522,37 @@ public void accumulate(Map<Long, PdfTemplate> container, RowView view) {
     class SubstitutionsReducer implements LinkedHashMapRowReducer<Long, PdfSubstitution> {
         @Override
         public void accumulate(Map<Long, PdfSubstitution> container, RowView view) {
-            SubstitutionType type = SubstitutionType.valueOf(view.getColumn("substitution_type", String.class));
-            PdfSubstitution sub;
-            switch (type) {
-                case PROFILE:
-                    sub = view.getRow(ProfileSubstitution.class);
-                    break;
-                case ACTIVITY_DATE:
-                    sub = view.getRow(ActivityDateSubstitution.class);
-                    break;
-                case ANSWER:
-                    QuestionType qtype = QuestionType.valueOf(view.getColumn("a_question_type", String.class));
-                    if (qtype == QuestionType.BOOLEAN) {
-                        sub = view.getRow(BooleanAnswerSubstitution.class);
-                    } else {
-                        sub = view.getRow(AnswerSubstitution.class);
-                    }
-                    break;
-                default:
-                    throw new DaoException("unhandled pdf substitution type " + type);
+            long substId = view.getColumn("p_pdf_substitution_id", Long.class);
```

Comment:
They're the same thing, but for consistency I think we should use column `a_pdf_substitution_id` here.

Reply:
Agree

Reply:
Hm, `a_pdf_substitution_id` is null for substitutions without `fields`

Reply:
@yufengwng What do you think?

Reply:
Wait, I'm not understanding why it's null when it's coming from the same column as `p_pdf_substitution_id`?

Reply:
My bad, I misunderstood the point. Fixed